src/app/shared/services/back-navigation.service.ts

```diff
@@ -1,25 +0,0 @@
-import { Injectable } from '@angular/core';
-import { Location } from '@angular/common';
-import { NavigationEnd, Router } from '@angular/router';
-
-@Injectable({providedIn: 'root'})
-export class BackNavigationService {
-    private history: string[] = [];
-
-    constructor(private router: Router, private location: Location) {
-        this.router.events.subscribe((event) => {
-            if (event instanceof NavigationEnd) {
-                this.history.push(event.urlAfterRedirects);
-            }
-        });
-    }
-
-    back(): void {
-        this.history.pop();
-        if (this.history.length > 0) {
-            this.location.back();
-        } else {
-            this.router.navigateByUrl('/');
-        }
-    }
-}
```

Comment:
We don't need a service for this. 
History is going to be filled up to infinity. Moreover, the same history functionality has been implemented by Angular Location service, so we can use it.